pkg/defaults/defaults.go

```diff
@@ -803,3 +812,34 @@ func buildRootImageStreamFromRepository(readFile readFile) (*api.ImageStreamTagR
 	}
 	return &config.BuildRootImage, nil
 }
+
+func ensureImageStreamTag(ctx context.Context, client ctrlruntimeclient.Client, isTagRef *api.ImageStreamTagReference, second time.Duration) {
+	istImport := &testimagestreamtagimportv1.TestImageStreamTagImport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      isTagRef.Name + "-" + isTagRef.Tag,
+			Namespace: "ci",
+		},
+		Spec: testimagestreamtagimportv1.TestImageStreamTagImportSpec{
+			Namespace: isTagRef.Namespace,
+			Name:      isTagRef.Name + ":" + isTagRef.Tag,
+		},
+	}
+	istImport.SetImageStreamLabels()
+
+	// Conflicts are expected
+	if err := client.Create(ctx, istImport); err != nil && !kapierrors.IsConflict(err) {
+		logrus.WithError(err).Warnf("Failed to create imagestreamtagimport for root %s", istImport.Name)
+	}
+
+	if err := wait.PollImmediate(5*second, 30*second, func() (bool, error) {
```

Comment:
If this is always going to use a duration of `second`, do we need to pass that in as a parameter?

Reply:
It doesn't, the tests pass `time.Nanosecond` as "second" into `stepConfigsForBuild`, otherwise we'd have to wait number of unittests * unittestsrepetitions (25 in CI, to avoid flakes) * 30 seconds for the tests, which makes CI timeout

Reply:
Ah. I missed that the test passed in something different, sorry!